monkey/common/types/plugin_name.py

```diff
@@ -0,0 +1,13 @@
+import re
+
+from pydantic import ConstrainedStr
+
+
+class PluginName(ConstrainedStr):
+    """
+    Plugin name in snake_case
+    """
+
+    strip_whitespace = True
+    to_lower = True
+    regex = re.compile("^[a-zA-Z0-9_]+$")
```

Comment:
Why are we doing `to_lower` but allowing `A-Z` in the regex? I see no reason to limit this to lowercase characters.

```suggestion
    regex = re.compile("^[a-zA-Z0-9_]+$")
```

Reply:
regex is validation, to_lower is processing. What happens if we have two plugins SSH and ssh? Are they different? Maybe it's worth allowing A-Z, but making sure that the name is unique